Add plugin bibtex-scholar to community-plugins.json

liu-qilong opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/liu-qilong/bibtex-scholar

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
["fundingUrl": "",](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/manifest.json#L9)
This is meant for links to services like _Buy me a coffee_, _GitHub sponsors_ and so on, if you don't have such a link remove this line.

[counter-reset: katexEqnNo mmlEqnNo;](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/styles.css#L1164)
Do not use `counter-reset` on `body`. It's bound to conflict with other plugins, as only the "last in" `counter-reset` will be applied. And, unless your elements that reference these counters are direct children of `body`, the counters will not increment.

['Copy All BibTeX'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L39), ['Copy All BibTeX Entries'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L45), ['Fetch BibTeX Online'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L108), ['Fetch BibTeX Online'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L114), ['Paper Panel'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L132), ['Open Paper Panel'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L138), ['Fetch BibTeX Online'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/bibtex.ts#L279), ['ID Surfix'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/bibtex.ts#L300), ['Paper Panel'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/panel.ts#L26)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[checkCallback: (checking: boolean) => {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L46), [checkCallback: (checking: boolean) => {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L70), [checkCallback: (checking: boolean) => {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L115), [checkCallback: (checking: boolean) => {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L139)
This should just be `callback` since there is no condition for this command.

[checkCallback: (checking: boolean) => {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L58-L63), [checkCallback: (checking: boolean) => {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L83-L93)
This should be checking for if there's a current file, if there's not, return `false`.
https://docs.obsidian.md/Plugins/User+interface/Commands#Conditional+commands

[this.plugin.cache.note_folder = value](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L397-L398), [this.plugin.cache.pdf_folder = value](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L407-L408)
[Use `normalizePath` to clean up user defined paths](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+%60normalizePath()%60+to+clean+up+user-defined+paths)

[new Setting(containerEl)](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L391-L399), [new Setting(containerEl)](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L401-L409)
Use [AbstractInputSuggest](https://docs.obsidian.md/Reference/TypeScript+API/AbstractInputSuggest) when letting a user select a file/folder to add type-ahead support.

[MarkdownRenderer.render(this.app, \`Warning: BibTeX ID has been used\n\`${id}\`\nRevise for successful import\`, p, '', this)](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L192)
The component passed to this function should not be the plugin instance, it's too long lived, it should only live as long as necessary, otherwise this leads to a memory leak.
You should create a new class that extends the [`MarkdownRenderChild`](https://docs.obsidian.md/Reference/TypeScript+API/MarkdownRenderChild) class and add it to the context of the MarkdownCodeblockProcessor using [`ctx.addChild()`](https://docs.obsidian.md/Reference/TypeScript+API/MarkdownPostProcessorContext/addChild).

[if (!await app.vault.adapter.exists(folder)) {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/hover.tsx#L115), [if (!await this.app.vault.adapter.exists(this.folder)) {](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/hover.tsx#L65)
[Prefer the Vault API over the Adapter API](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Prefer+the+Vault+API+over+the+Adapter+API)

---

Optional feedback:

[// events for rename and delete file](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/main.ts#L96-L103)
Should you also account for [create](https://docs.obsidian.md/Reference/TypeScript+API/Vault/on('create')) and [modify](https://docs.obsidian.md/Reference/TypeScript+API/Vault/on('modify'))? And potentially initialize a cache on startup after the [layout is ready](https://docs.obsidian.md/Reference/TypeScript+API/Workspace/onLayoutReady0?)

Zachatoo commented:
The bot hasn't detected changes for these requested changes. Once you make these changes then the bot will push this back into my queue within 4 hours.

['Paper Panel'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/panel.ts#L26), ['Fetch BibTeX Online'](https://github.com/liu-qilong/bibtex-scholar/blob/b83d377a6de06638ba1c4f49bb2d5885f9c215ed/src/bibtex.ts#L279)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

liu-qilong commented:
@Zachatoo Thanks for pointing those out! I’ve resolved the issues

Zachatoo commented:
[if (!checking) {](https://github.com/liu-qilong/bibtex-scholar/blob/c1ef3cb976f759b25795ae0c5831eae6e53f82b9/src/main.ts#L56-L66), [if (!checking) {](https://github.com/liu-qilong/bibtex-scholar/blob/c1ef3cb976f759b25795ae0c5831eae6e53f82b9/src/main.ts#L85-L100)
You shouldn't assume that because you aren't checking, that the command is valid. This is not true when executing a command that is bound to a hotkey. If the condition isn't true, you should always return early, even if you aren't checking.

[console.log(this)](https://github.com/liu-qilong/bibtex-scholar/blob/c1ef3cb976f759b25795ae0c5831eae6e53f82b9/src/prompt.ts#L93)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

---

Optional feedback:

[MarkdownRenderer.render(this.app, \`Warning: BibTeX ID has been used\n\`${id}\`\nRevise for successful import\`, p, '', component)](https://github.com/liu-qilong/bibtex-scholar/blob/c1ef3cb976f759b25795ae0c5831eae6e53f82b9/src/main.ts#L196)
You aren't rendering any Markdown, just plain text. You can simplify this by setting this text as the child of the `p` tag and rendering that directly. Is there a reason you specifically want to use the Markdown renderer for this?

liu-qilong commented:
@Zachatoo Thanks for your feedback! I misunderstood the logic of [checkCallback](https://docs.obsidian.md/Reference/TypeScript+API/Command/checkCallback). I have fixed it accordingly. The notice was also changed from markdown renderer to plain text
